isc_dhcp_leases/iscdhcpleases.py

```diff
@@ -160,6 +162,9 @@ def __init__(self, ip, data, cltt, host_identifier, address_type):
         self.last_communication = cltt
 
         self.host_identifier = codecs.decode(host_identifier, "unicode_escape")
+        host_identifier_bytes = self.host_identifier.encode('latin-1')
```

Comment:
It might be better to use the host_identifier bytes before the decoding on line 164.